### PR TITLE
fix(types): add to cast to projection alias

### PR DESCRIPTION
**Pull request by lohart13**
## Pull Request Checklist
- [x] Have you added new tests to prevent regressions?
- [ ] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [x] Did you update the typescript typings accordingly (if applicable)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change

Currently the cast function is missing from the projection alias which means you are unable to use cast in the attributes of a find query.
This PR adds the cast function to the list of possible options.

### CODE REVIEW

**Review by ephys — DISMISSED** [review]
Nice work :)

File: test/unit/query-generator/select-query.test.ts [hunk 1]
```diff
@@ -75,6 +75,25 @@ describe('QueryGenerator#selectQuery', () => {
...
+      ],
+    }, User);
+
+    expectsql(sql, {
+      default: 'SELECT `id`, CAST(`createdAt` AS VARCHAR) AS `createdAt` FROM `Users` AS `User`;',
+      mssql: `SELECT [id], CAST([createdAt] AS VARCHAR) AS [createdAt] FROM [Users] AS [User];`,
+      postgres: `SELECT "id", CAST("createdAt" AS VARCHAR) AS "createdAt" FROM "Users" AS "User";`,
+      snowflake: `SELECT "id", CAST("createdAt" AS VARCHAR) AS "createdAt" FROM "Users" AS "User";`,
+      ibmi: `SELECT "id", CAST("createdAt" AS VARCHAR) AS "createdAt" FROM "Users" AS "User"`,
+      db2: `SELECT "id", CAST("createdAt" AS VARCHAR) AS "createdAt" FROM "Users" AS "User";`,
```
Comment (WikiRik) [inline test/unit/query-generator/select-query.test.ts:93]:
```suggestion
      default: `SELECT [id], CAST([createdAt] AS VARCHAR) AS [createdAt] FROM [Users] AS [User];`,
```
This should work

**Review by WikiRik — APPROVED** [review]
We can make the test easier, but apart from that this is good. Thanks for adding an unit test as well

**Comment by WikiRik** [comment]
I do have one additional remark for all your PRs; can you enable the 'Allow edits from maintainers' so we can merge main in this?

**Comment by lohart13** [comment]
> I do have one additional remark for all your PRs; can you enable the 'Allow edits from maintainers' so we can merge main in this?

@WikiRik I've enabled this on my PR's now

**Comment by lohart13** [comment]
> We can make the test easier, but apart from that this is good. Thanks for adding an unit test as well

I've added the commit and merged main into this PR.

**Review by WikiRik — APPROVED** [review]

**Pull request merged**